WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController+Extensions.swift

```diff
@@ -478,6 +478,202 @@ extension NotificationDetailsViewController
 
 
 
+/// Resources
+///
+extension NotificationDetailsViewController
+{
+    func displayURL(url: NSURL?) {
+        guard let url = url else {
+            tableView.deselectSelectedRowWithAnimation(true)
+            return
+        }
+
+        // Attempt to infer the NotificationRange associated: Recover Metadata + Push Native Views!
+        //
+        do {
+            let range = note.notificationRangeWithUrl(url)
+            try displayResourceWithRange(range)
+        } catch {
+            displayWebViewWithURL(url)
+        }
+    }
+
+    func displayNotificationSource() {
+        guard let type = note.type, let resourceURL = note.resourceURL() else {
+            tableView.deselectSelectedRowWithAnimation(true)
+            return
+        }
+
+        do {
+            switch type {
+            case NoteTypeFollow:
+                try displayStreamWithSiteID(note.metaSiteID)
+
+            case NoteTypeLike:
+                fallthrough
+
+            case NoteTypeMatcher:
+                fallthrough
+
+            case NoteTypePost:
+                try displayReaderWithPostId(note.metaPostID, siteID: note.metaSiteID)
+
+            case NoteTypeComment:
+                fallthrough
+
+            case NoteTypeCommentLike:
+                try displayCommentsWithPostId(note.metaPostID, siteID: note.metaSiteID)
+
+            default:
+                throw DisplayError.UnsupportedType
+            }
+        } catch {
+            displayWebViewWithURL(resourceURL)
+        }
+    }
+
+
+
+    // MARK: - Displaying Ranges!
+
+    private func displayResourceWithRange(range: NotificationRange?) throws {
+        guard let range = range else {
+            throw DisplayError.MissingParameter
+        }
+
+        switch range.type {
+        case .Site:
+            try displayStreamWithSiteID(range.siteID)
+
```

Comment:
For consistency's sake, can we remove the spacing between the case statements please? None of the other switch statements in the app have spaces between them.